src/docfx/build/xref/InternalXrefSpec.cs

```diff
@@ -21,52 +27,84 @@ internal class InternalXrefSpec : IXrefSpec
+        public string GetName() => GetXrefPropertyValueAsString("name");
 
         public ExternalXrefSpec ToExternalXrefSpec(Context context, bool forXrefMapOutput)
         {
-            var spec = new ExternalXrefSpec
+            if (t_recursionDetector.Value.Contains((Uid, DeclaringFile)))
             {
-                Uid = Uid,
-                Monikers = Monikers,
-            };
+                var referenceMap = t_recursionDetector.Value.Select(x => x.parent).ToList();
+                throw Errors.CircularReference(referenceMap).ToException();
+            }
 
-            if (forXrefMapOutput)
+            try
             {
-                var (_, _, fragment) = UrlUtility.SplitUrl(Href);
-                var path = DeclaringFile.CanonicalUrlWithoutLocale;
+                t_recursionDetector.Value.Push((Uid, DeclaringFile));
+                var spec = new ExternalXrefSpec
+                {
+                    Uid = Uid,
+                    Monikers = Monikers,
+                };
 
-                // DHS appends branch infomation from cookie cache to URL, which is wrong for UID resolved URL
-                // output xref map with URL appending "?branch=master" for master branch
-                var query = DeclaringFile.Docset.Repository?.Branch == "master" ? "?branch=master" : "";
-                spec.Href = path + query + fragment;
-            }
-            else
-            {
-                // relative path for internal UID resolving
-                spec.Href = PathUtility.GetRelativePathToFile(DeclaringFile.SiteUrl, Href);
-            }
+                if (forXrefMapOutput)
+                {
+                    var (_, _, fragment) = UrlUtility.SplitUrl(Href);
+                    var path = DeclaringFile.CanonicalUrlWithoutLocale;
 
-            foreach (var (key, value) in ExtensionData)
-            {
-                try
+                    // DHS appends branch infomation from cookie cache to URL, which is wrong for UID resolved URL
+                    // output xref map with URL appending "?branch=master" for master branch
+                    var query = DeclaringFile.Docset.Repository?.Branch == "master" ? "?branch=master" : "";
+                    spec.Href = path + query + fragment;
+                }
+                else
                 {
-                    spec.ExtensionData[key] = value.Value;
+                    // relative path for internal UID resolving
+                    spec.Href = PathUtility.GetRelativePathToFile(DeclaringFile.SiteUrl, Href);
                 }
-                catch (DocfxException ex)
+
+                foreach (var (key, value) in ExtensionData)
                 {
-                    context.ErrorLog.Write(DeclaringFile, ex.Error);
+                    try
+                    {
+                        spec.ExtensionData[key] = value.Value;
+                    }
+                    catch (DocfxException ex)
```

Comment:
could this catch be the cause of the problem?

Reply:
This is used to add `DeclaringFile` for the error, because while outputting xref map, there is no file attached to the error